packages/types/src/type/Vote.ts

```diff
@@ -2,41 +2,78 @@
 // This software may be modified and distributed under the terms
 // of the Apache-2.0 license. See the LICENSE file for details.
 
-import { isBoolean } from '@polkadot/util';
+import { isBoolean, isNumber, isObject, isU8a } from '@polkadot/util';
 
-import I8 from '../primitive/I8';
+import Conviction from './Conviction';
+import U8a from '../codec/U8a';
+import Boolean from '../primitive/Bool';
 
 /**
  * @name Vote
  * @description
  * A number of lock periods, plus a vote, one way or the other.
  */
-export default class Vote extends I8 {
+export default class Vote extends U8a {
+  private _aye: Boolean;
+  private _conviction?: Conviction;
+
   constructor (value?: any) {
-    super(Vote.decodeVote(value));
+    // decoded is just 1 byte
+    // Aye: Most Significant Bit
+    // Conviction: 0000 - 0101
+    const decoded = Vote.decodeVote(value);
+
+    super(decoded);
+
+    const msb = decoded[0] >> 7;
+    const conviction = decoded[0] & 0b01111111;
+
+    this._aye = new Boolean(msb);
+    this._conviction = new Conviction(conviction);
   }
 
-  private static decodeVote (value?: any): any {
+  private static decodeVote (value?: any): Uint8Array {
     if (isBoolean(value)) {
-      return value ? -1 : 0;
+      return value ? new Uint8Array([0b10000000]) : new Uint8Array([0b0]);
     } else if (value instanceof Boolean) {
       return Vote.decodeVote(value.valueOf());
+    } else if (isNumber(value) && value !== 0) {
+      return value < 0 ? new Uint8Array([0b10000000]) : new Uint8Array([0b0]);
+    } else if (isObject(value) && value.aye && value.conviction) {
```

Comment:
value.aye can be false, and value.conviction can be 0. we should use isUndefined here